test/test_parse_shebang.py

```diff
@@ -146,3 +148,27 @@ def test_throw_runtime_error_when_weird_object_passed(self):
         """Raise RuntimeError when weird object is passed in."""
         with ExpectedException(RuntimeError):
             parse(object())
+
+    # suppress(no-self-use)
+    def test_unicode(self):
+        """A filename can be text which in python 2 is not `str`."""
+        tmpdir = mkdtemp()
+        self.addCleanup(shutil.rmtree, tmpdir)
+
+        filename = os.path.join(tmpdir, u"foo.sh")
+        with io.open(filename, "w") as f:
```

Comment:
Indeed that is correct.  Whenever I write 2+3 code I use the `io` module so I'm always dealing with the same types on both sides of the fence